lang/java/compiler/src/main/javacc/org/apache/avro/compiler/idl/idl.jj

```diff
@@ -1444,15 +1443,42 @@ Schema UnannotatedType(Map<String, JsonNode> props):
 {
   Schema s;
 }
+{
+  (
+    s = NullableType(props)
```

Comment:
This is what makes the magic possible:
1. Primitive/logical & reference types are handled differently (elsewhere)
2. The other types are handled as usual